presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1615,4 +1619,40 @@ protocol::NodeStatus PrestoServer::fetchNodeStatus() {
   return nodeStatus;
 }
 
+void PrestoServer::registerDynamicFunctions() {
+  auto systemConfig = SystemConfig::instance();
+  if (!systemConfig->pluginDir().empty()) {
+    const fs::path path(systemConfig->pluginDir());
+    PRESTO_STARTUP_LOG(INFO) << "Dynamic library loading path: " << path;
+    std::error_code ec;
+    if (fs::is_directory(path, ec)) {
+      const fs::path configPath(systemConfig->dynamicLibraryValidatorConfig());
+      if (!fs::exists(configPath, ec)) {
+        LOG(ERROR)
+            << "Config file not found in path: " << configPath
+            << ". Dynamic libraries will not be loaded on this worker run.";
+        return;
+      } else {
+        DynamicLibraryValidator dVal(configPath, systemConfig->pluginDir());
+        auto filenameAndEntrypointMap = dVal.getEntrypointMap();
+        auto registeredFnSignaturesBefore = velox::getFunctionSignatures();
+        for (const auto& entryPointItr : filenameAndEntrypointMap) {
```

Comment:
All this logic can be moved into the DynamicLibraryValidator class (maybe call it DynamicLibraryLoader instead).

The code here would only read the config entries for plugin.dir and dynamiclibraryvalidator and pass them to DynamicLibraryLoader.